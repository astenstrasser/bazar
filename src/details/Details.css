.loading {
  margin: 3rem;
  text-align: center;
  font-size: 22px;
  font-weight: 400;
  color: #023047;
}

.item {
  display: flex;
  flex-direction: column;
}

@media (max-width: 600px) {
  .item--body {
    display: flex;
    flex-direction: column;
  }
  .item--information {
    margin-top: 2rem;
    margin-bottom: 2rem;
  }
  .item {
    margin: 1rem;
  }
}

@media (min-width: 600px) {
  .item {
    justify-content: center;
    align-items: center;
  }
  .details-page {
    margin: 4rem 7rem;
  }
}

.item--title {
  width: 75%;
  color: #003559;
}

.item--body {
  display: flex;
}

.item--picture {
  max-width: 30rem;
  border: solid 3rem #f7f7f7;
}

.item--information {
  background-color: #f7f7f7;
  display: flex;
  flex-direction: column;
  padding: 2rem 10rem 0 1rem;
  color: #003559;
}

.item--information span {
  font-size: 28px;
  font-weight: 600;
  color: #ffb703;
}

.return--button {
  text-decoration: none;
  color: #003559;
  font-weight: 600;
  display: flex;
  align-items: center;
}

.return--button img {
  width: 32px;
}
