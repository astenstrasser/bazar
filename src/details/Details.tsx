import React, { useEffect, useState } from "react";
import { useGoogleAuth } from "react-gapi-auth2";
import { useHistory, useParams } from "react-router-dom";
import "./Details.css";
import { Product } from "../common/types";
import LoadingTag from "../common/loading-tag/LoadingTag";
import ActionIcon, { Icon } from "../common/action-icon/ActionIcon";

function Details(): React.ReactElement {
  const { id }: { id: string } = useParams();

  const { googleAuth } = useGoogleAuth();
  console.log("user logado", googleAuth?.isSignedIn.get());

  const history = useHistory();
  const [product, setProduct] = useState<Product>();
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    const url = `https://bah-zar-api.herokuapp.com/product/${id}`;
    isLoading &&
      fetch(url)
        .then((res) => res.json())
        .then((product: Product) => {
          setProduct(product);
          setIsLoading(false);
        });
    return () => setIsLoading(false);
  }, [LoadingTag]);

  return (
    <div className="details-page">
      <LoadingTag isLoading={isLoading} />
      {product && (
        <div className="item">
          <h1 className="item--title">{product.name.toUpperCase()}</h1>
          <div className="item--body">
            <img
              className="item--picture"
              src={product?.imageUrl}
              alt={product?.imageAltText}
            />
            <div className="item--information">
              <span> R$ {product.price}</span>
              <p>
                <strong>Descrição:</strong>
                <br />
                {product.details}
              </p>
            </div>
          </div>
        </div>
      )}
      <ActionIcon
        onClick={() => {
          history.push("/");
        }}
        alt="Seta para retornar à página anterior"
        type={Icon.LeftArrow}
      />
    </div>
  );
}

export default Details;
