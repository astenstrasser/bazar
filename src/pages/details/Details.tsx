import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import "./Details.css";
import { Product } from "../../shared-components/types";
import LoadingTag from "../../shared-components/loading-tag/LoadingTag";
import ReturnIcon from "../../shared-components/return-icon/ReturnIcon";
import { fetchProductById } from "../../api/api";

function Details(): React.ReactElement {
  const { id }: { id: string } = useParams();

  const [product, setProduct] = useState<Product>();
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    isLoading &&
      fetchProductById(id)
        .then((res) => res.json())
        .then((product: Product) => {
          setProduct(product);
          setIsLoading(false);
        });
    return () => setIsLoading(false);
  }, [LoadingTag]);

  return (
    <div className="details-page">
      <LoadingTag isLoading={isLoading} />
      {product && (
        <div className="item">
          <h1 className="item--title">{product.name.toUpperCase()}</h1>
          <div className="item--body">
            <img
              className="item--picture"
              src={product?.imageUrl}
              alt={product?.imageAltText}
            />
            <div className="item--information">
              <span> R$ {product.price}</span>
              <p>
                <strong>Descrição:</strong>
                <br />
                {product.details}
              </p>
            </div>
          </div>
        </div>
      )}
      <ReturnIcon reference={"/"} />
    </div>
  );
}

export default Details;
