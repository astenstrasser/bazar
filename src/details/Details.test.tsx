import React from "react";
import { render } from "@testing-library/react";
import Details from "./Details";

test("renders snapshot", () => {
  const { container } = render(<Details />);
  expect(container.firstChild).toMatchSnapshot();
});
