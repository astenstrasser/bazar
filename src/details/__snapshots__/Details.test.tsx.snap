// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders product 1`] = `
<div>
  <div
    class="details-page"
  >
    <div />
    <div
      class="item"
    >
      <h1
        class="item--title"
      >
        MOCK PRODUCT 1
      </h1>
      <div
        class="item--body"
      >
        <img
          class="item--picture"
          src="http://localhost:8080/image"
        />
        <div
          class="item--information"
        >
          <span>
             R$ 
            100
          </span>
          <p>
            <strong>
              Descrição:
            </strong>
            <br />
          </p>
        </div>
      </div>
    </div>
    <a>
      <img
        alt="Seta para retornar à página anterior"
        class="action-icon"
        src="left-arrow.svg"
      />
    </a>
  </div>
</div>
`;

exports[`renders snapshot while loading 1`] = `
<div>
  <div
    class="details-page"
  >
    <div>
      <div
        class="loading-container"
      >
        <div
          class="spinner"
        >
          <div
            class="petal"
            id="p1"
          />
          <div
            class="petal"
            id="p2"
          />
          <div
            class="petal"
            id="p3"
          />
          <div
            class="petal"
            id="p4"
          />
        </div>
        <span>
          Loading...
        </span>
      </div>
    </div>
    <a>
      <img
        alt="Seta para retornar à página anterior"
        class="action-icon"
        src="left-arrow.svg"
      />
    </a>
  </div>
</div>
`;
